### PR TITLE
Closes #26185: add telemetry events for synced history screen

### LINKED ISSUE #26185: Add Telemetry to Synced History Screen
Relates to [this](https://github.com/mozilla-mobile/fenix/pull/25943)

The required telemetry would be:
– How often does someone open the synced history view;
– How often does someone open a synced history item.

The ideal telemetry for synced history would copy the history events
https://dictionary.telemetry.mozilla.org/apps/fenix?page=1&search=history

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21113)

### PR DESCRIPTION
**Pull request by mavduevskiy**
Closes #26185 

draft 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26185

### CODE REVIEW

**Comment by mavduevskiy** [comment]
@rocketsroger 
May be instead of creating new events for synced history fragment, we could pass extras to the existing history events to distinguish the source?

File: app/metrics.yaml [hunk 1]
```diff
@@ -3244,6 +3244,247 @@ history:
...
+    notification_emails:
+      - android-probes@mozilla.com
+    expires: 117
+    metadata:
+      tags:
+        - History
+  removed_all:
+    type: event
+    description: |
+      A user removed all history items
```
Comment (rocketsroger) [inline app/metrics.yaml:3283]: This is removing all history items? or just Synced history items?

File: app/metrics.yaml [hunk 2]
```diff
@@ -3244,6 +3244,247 @@ history:
...
 
+synced_history:
+  opened:
+    type: event
+    description: |
+      A user opened the synced history screen
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/26185
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/26185
```
Comment (rocketsroger) [inline app/metrics.yaml:3255]: This should point to this PR, not the issue.  Thanks

**Comment by rocketsroger** [comment]
> @rocketsroger May be instead of creating new events for synced history fragment, we could pass extras to the existing history events to distinguish the source?

As discussed, let's clarify the requirements for the delete and search buttons when showing synced history.  If the requirement is not to show those buttons, then we can just create the events without using extras.  I lean towards not using extras to multiplex the history telemetry events because: 1. This will effect history telemetry queries. 2. If we separate the implementation of history fragment vs synced history fragment in the future then sharing telemetry events won't be as useful.

**Pull request closed without merging**